corporate/lib/stripe.py

```diff
@@ -1170,6 +1189,50 @@ def set_required_plan_tier(self, required_plan_tier: int) -> str:
             plan_tier_name = CustomerPlan.name_from_tier(new_plan_tier)
         return f"Required plan tier for {self.billing_entity_display_name} set to {plan_tier_name}."
 
+    def create_fixed_price_plan(self, fixed_price: int) -> str:
+        customer = self.get_customer()
+        if customer is None:
+            customer = self.update_or_create_customer()
+
+        if customer.required_plan_tier is None:
+            raise SupportRequestError("Required plan tier should not be set to None")
+        required_plan_tier_name = CustomerPlan.name_from_tier(customer.required_plan_tier)
+
+        fixed_price_cents = fixed_price * 100
+        plan = get_current_plan_by_customer(customer)
+        if plan is not None:
+            fixed_price_plan = CustomerPlan.objects.create(
+                customer=customer,
+                automanage_licenses=True,
+                charge_automatically=plan.charge_automatically,
+                fixed_price=fixed_price_cents,
+                # 'billing_cycle_anchor' will be decided during /upgrade.
+                billing_cycle_anchor=timezone_now() - timedelta(weeks=52),
```

Comment:
Why this specific choice for the unused value?

Reply:
We need a dummy value here as we can't create a plan without `billing_cycle_anchor`. The actual value would depend on whether the user buys a plan or schedules it to switch after end of legacy plan.

Reply:
I think, we should use `datetime.utcfromtimestamp(0)` as the dummy value instead of a value which is possible.

Reply:
That is probably better; making the field nullable is another option. Maybe start a chat.zulip.org thread about this detail?

Reply:
[CZO thread](https://chat.zulip.org/#narrow/stream/273-kandra-billing/topic/Make.20.60billing_cycle_anchor.60.20nullable/near/1725674)

Reply:
We're now following the `CustomerPlanOffer` table approach.